src/gui/IconDownloader.cpp

```diff
@@ -91,16 +99,15 @@ void IconDownloader::setUrl(const QString& entryUrl)
     // searching for a match with the returned address(es).
     bool hostIsIp = false;
     QList<QHostAddress> hostAddressess = QHostInfo::fromName(fullyQualifiedDomain).addresses();
-    for (const auto& addr : hostAddressess) {
-        if (addr.toString() == fullyQualifiedDomain) {
-            hostIsIp = true;
-        }
-    }
+    hostIsIp =
+        std::any_of(hostAddressess.begin(), hostAddressess.end(), [&fullyQualifiedDomain](const QHostAddress& addr) {
+            return addr.toString() == fullyQualifiedDomain;
+        });
 
     // Determine the second-level domain, if available
     QString secondLevelDomain;
     if (!hostIsIp) {
-        secondLevelDomain = getSecondLevelDomain(m_url);
+        secondLevelDomain = getSecondLevelDomain(url);
```

Comment:
Does this preserve the port number? Should it?

Reply:
It preserves port number, username/password, query, ..., i.e., all non-host fields of the URL. I see the issue here, still, i think we should at least consider the URL as a candidate. Perhaps adding two URLs, one blank request to the second level domain with the original query and one that keep all original data intact would be reasonable?

Reply:
Yes I agree, that is in the spirit of maximizing success. Generally speaking this should never be hit since it is rare that a subdomain does not provide its own icon.